rxandroidble/src/test/groovy/com/polidea/rxandroidble/RxBleClientTest.groovy

```diff
@@ -315,4 +313,31 @@ class RxBleClientTest extends Specification {
         mock.getAddress() >> scanData['deviceMac']
         bleAdapterWrapperSpy.addScanResult(mock, scanData['rssi'], scanData['scanRecord'])
     }
+
+    def "should not call stopLeScan if startLeScan wasn't called before"() {
```

Comment:
We should have a test that is currently causing us problems. Something like `def "should not startLeScan() if stop() was called before"`. Maybe it will be even easier.